data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodeCertRenewalPeriodical.java

```diff
@@ -98,7 +99,13 @@ private boolean needsNewCertificate(RenewalPolicy renewalPolicy) {
     }
 
     private boolean expiresSoon(Date expiration, RenewalPolicy renewalPolicy) {
-        final Duration threshold = Duration.parse(renewalPolicy.certificateLifetime()).dividedBy(CERT_RENEWAL_THRESHOLD_PERCENTAGE);
+        Duration threshold;
+        PeriodDuration lifetime = PeriodDuration.parse(renewalPolicy.certificateLifetime());
```

Comment:
good point. I will move it

Reply:
added an extra method which is used to avoid having to write migrations and changes in the resource which receives a string from the frontend component